### DISCUSSION TITLE
Not able to understand how to install the script

### DISCUSSION THREAD

**Discussion by Welstand** [lucasrla/remarks#23] [Q&A]
Hi, would it be possible to make a Wiki for people like me who don't understand _anything_ of programming? I really want/need to use the script, so it's quite frustrating not understanding how to do it.

**Comment by akelai** [comment]
For which OS you’re most interested in?

**Comment by Welstand** [comment]
macOS Big Sur Version 11.2.2

**Comment by lucasrla** [comment]
Hey @Welstand,

To run `remarks`, you need just two things: `git` and `python`. Python 3.8 comes pre-installed with macOS Big Sur. So the only new thing you need is git. There are several ways to install it. A straightforward one is described here: https://docs.github.com/en/github/getting-started-with-github/set-up-git

There is a catch with Apple's default Python, though. [Apple ships both `python2` and `python3` on macOS Big Sur](https://apple.stackexchange.com/questions/406244/will-macos-big-sur-remove-the-default-python-installation-for-good). Because `remarks` needs Python `>= 3.8` you must use the command `python3` and `pip3` and not simply `python` and `pip` (which unfortunately point to `python2`).

Now, just open your terminal and type the following:

```sh
# Download remarks to your machine
git clone https://github.com/lucasrla/remarks.git && cd remarks

# Install the dependencies
pip3 install -r requirements.txt

# Run it and learn the features that are available
python3 -m remarks --help
```

Please let us know if you run into any problems.

PS: I have just enabled the Wiki feature. It is open for anyone to edit/write it: https://github.com/lucasrla/remarks/wiki. Feel free to document your steps there so that future users can learn as well.

PPS: I will convert this issue to a discussion. It seems more appropriate there.

**Comment by Welstand** [comment]
Thank you for your help! Now, (I guess) the script is installed on my computer. Though, I have still problems with the setting up. In the guide you write: 

> Using these credentials, scp the contents of /home/root/.local/share/remarkable/xochitl from your reMarkable to a directory on your computer

What does it mean, to "scp" the contents of "/home/root/.local/share/remarkable/xochitl"? Do you know of any down-to-earth guide of scp?

**Reply by lucasrla** [reply]
Nice!

Have you been able to run the demo (Turing's paper)? If yes, `remarks` has been properly installed on your computer.

Re your question about `scp`, I have just updated and expanded the [Setup instructions](https://github.com/lucasrla/remarks#setup). It should be clearer now that there are at least four options for copying to your computer the internal files that `remarks` needs. The most convenient option (and the path I recommend that you take) is to simply leverage reMarkable's own desktop application as described there.

Good luck! 

And please don't hesitate to share here any other question you might have.

**Comment by Welstand** [comment]
Thanks again for helping! Thanks to the new wiki, I guess I managed to install remarks. I reached now the section "Usage and Demo". After digiting `python -m remarks --help` in the terminal, I received this message: `/System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python: No module named remarks`.

**Reply by lucasrla** [reply]
You are getting this error because Apple defaults to Python 2.7. 

As I have written above, for your setup, you must **always** use the command `python3` to get `remarks` to work. So, do this instead:

```sh
python3 -m remarks --help
```

**Reply by Welstand** [reply]
After digitng `python3 -m remarks --help`, I get this message: `/Library/Frameworks/Python.framework/Versions/3.9/bin/python3: No module named remarks.__main__; 'remarks' is a package and cannot be directly executed`

**Reply by lucasrla** [reply]
Let's troubleshoot this! 🙏 

Note that the [Setup instructions](https://github.com/lucasrla/remarks#setup) starts with `git clone ... & cd remarks`. My hypothesis is you are missing the need to `cd remarks` any time you want to run it.

One is only able to run `remarks` if it's called from inside the directory where it was git-cloned to. It looks like you are running `python3 -m remarks --help` from its parent directory, not from `remarks` own directory.

You probably ran `git clone https://github.com/lucasrla/remarks.git`, which has cloned remarks into, say, `/Users/Welstand/remarks`. Now, anytime you want to run it, you must switch to that directory in order to "access" `remarks`. Only in there `python3 -m remarks --help` will work properly. To enter that directory, simply type `cd remarks` and then run it with `python3 -m remarks --help`. If you don't switch the directory, it is as if are running remarks from `/Users/Welstand`, and not the correct `/Users/Welstand/remarks`. And that is probably why you are getting that error.

(I hope I explained it clearly enough.)

---

Additionally, in case my hypothesis is wrong and there is something else going on, can you please also enter the following sequence of commands on Terminal and paste their outputs here? 

(The lines starting with `#` are just my comments for you to learn what is going on, no need to enter them.)

```sh
# let's figure out what is your current directory path
pwd

# let's check if remarks has been cloned into this directory
ls | grep remarks

# let's check if remarks is available at /remarks
ls remarks

# let's check if git is really installed and properly working
git --version

# let's check what packages you have installed via pip3
pip3 list
```
